lexicon/__main__.py

```diff
@@ -8,10 +8,12 @@
 import os
 import sys
 import json
+import pkgutil
 
 import pkg_resources
 
-from .client import Client
+from lexicon.client import Client
+from lexicon import providers as providers_package
```

Comment:
this is where my python ignorance starts to show. Will this still work when using lexicon as a library vs an executable?

Reply:
In fact, this is even the best way to ensure that it will work outside the development area (so either as standalone executable or imported as library). In general, people say to use absolute import and relative import for corner situations (like if you need to dig into 10 levels of packages to get a module, but in this case, there is something to think about the library design ^^).

Because absolute import are not tied to the actual filesystem architecture of a given library.

The only thing to care about correctly is during development. Because in this case, the starting directory may influence whether a package is accessible or not. But you completely mitigate that if you set up your dev environnement "by the book", ie by invoking first `pip install -e lexicon_source_path`. Then lexicon package is correctly exposed, independently of its actual location, and you match your dev env with the deployed env. 

This what is advised in your CONTRIBUTING.md, so we are covered.